### ISSUE TITLE
Large config file - raw edit 500 error

### ISSUE THREAD

**Issue by MattCrum1** [sni/Thruk#1160]
**Describe the bug**
Thruk UI shows a 500 error when I raw edit a config file with >1000 lines.

Error in the apache ssl_error log is as follows:

`[Fri Feb 11 16:40:36.345930 2022] [fcgid:warn] [pid 376:tid 140265050244864] [client X.X.X.X:53494] mod_fcgid: HTTP request length 138275 (so far) exceeds MaxRequestLen (131072), referer: https://monitoring.customername/thruk/cgi-bin/conf.cgi?sub=objects&action=editor&file=/etc/naemon/objects/customer/customer_hosts.cfg&line=78&data.id=c2e14&back=edit`

Is this an Apache configuration which needs to be updated? Please let me know which one if so (it's running in a container and I can configure Apache in the Docker build process...).

**Thruk Version**
Thruk Version 2.46

**To Reproduce**
1. Go to Config page
2. Edit or paste 1000 lines of configuration
3. Save file - Thruk shows 500 error

**Expected behavior**
Thruk saves config file to FS ready for verification

**Desktop (please complete the following information):**
 - Firefox
 - Version 96.0.3 (64-bit)

**Comment by MattCrum1** [comment]
Think I've found the solution - we need to increase the MaxRequestLen parameter in the thruk.conf httpd module (/etc/httpd/conf.d/thruk.conf ):

```
<IfModule mod_fcgid.c>
FcgidMaxRequestLen 1073741824 
...
</IfModule>
```

Don't know if this needs updating in your documentation @sni  (or if it's there already and I missed it!)

**Comment by klaernie** [comment]
Taking a quick google I found this:
https://stackoverflow.com/questions/14066109/mod-fcgid-http-request-length-138520-so-far-exceeds-maxrequestlen-131072/18114711

It describes, that you should set [FcgidMaxRequestLen](https://httpd.apache.org/mod_fcgid/mod/mod_fcgid.html#fcgidmaxrequestlen) in your apache configuration to a value greater than the default 128KB. Maybe set it to 512KB (`524288`) and see if that's enough to edit your configuration.

**Comment by MattCrum1** [comment]
Hi @klaernie - thanks for your note - yeah I saw that and tried setting it in thruk.conf within /etc/httpd/conf.d but it didn't do anything, so I gave up.

Saw your response today and tried again - I enabled server-info and saw that the parameter wasn't being picked up by Apache. Other Fcgid parameters are configured in /etc/httpd/conf.d/fcgid.conf, so I added the line there and it worked OK - will close this off.
